read-fonts/src/tables/variations.rs

```diff
@@ -498,6 +502,277 @@ impl<'a> Iterator for TupleVariationHeaderIter<'a> {
     }
 }
 
+#[derive(Clone)]
+pub struct TupleVariationData<'a, T> {
+    pub(crate) axis_count: u16,
+    pub(crate) shared_tuples: Option<SharedTuples<'a>>,
+    pub(crate) shared_point_numbers: Option<PackedPointNumbers<'a>>,
+    pub(crate) tuple_count: TupleVariationCount,
+    // the data for all the tuple variation headers
+    pub(crate) header_data: FontData<'a>,
+    // the data for all the tuple bodies
+    pub(crate) serialized_data: FontData<'a>,
+    pub(crate) _marker: std::marker::PhantomData<T>,
```

Comment:
This always confuses me but I suspect this is not the param we want? This is telling rustc that we actually own an instance of `T`, which influences things like drop checking. Staring at https://doc.rust-lang.org/nomicon/phantom-data.html#table-of-phantomdata-patterns makes me think we want `PhantomData<fn() -> T>`? 🤮

Reply:
Good catch! I think I actually got caught up on this somewhere and had to add a lifetime bound.

Fixed